src/main.cc

```diff
@@ -12,14 +12,55 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <fstream>
 #include <iostream>
+#include <string>
 
 #include "class.h"
 
-int main(int argc, char *argv[]) {
+// Return true on success or false on failure.
+static bool ReadImage(const char filename[], WebPPicture* const pic) {
+  const uint8_t* data = NULL;
+  size_t data_size = 0;
+  WebPImageReader reader;
+  bool ok;
+
+  if (!ImgIoUtilReadFile(filename, &data, &data_size)) return false;
+  reader = WebPGuessImageReader(data, data_size);
+  ok = reader(data, data_size, pic, 1, NULL);
+
+  free((void*)data);
+  return ok;
+}
+
+int main(int argc, char* argv[]) {
   // TODO: read images given as arguments
   // TODO: create an animation
   // TODO: write the animation to disk
+
+  libwebp::Thumbnailer thumbnailer = libwebp::Thumbnailer();
+  WebPPicture pic;
+  WebPData webp_data;
+
+  WebPDataInit(&webp_data);
+  WebPPictureInit(&pic);
+
+  std::ifstream input_list(argv[1]);
+  const char* output = argv[2];
+  std::string filename_str;
+  int timestamp_ms;
+
+  while (input_list >> filename_str >> timestamp_ms) {
+    char filename[filename_str.length()];
+    strcpy(filename, filename_str.c_str());
+    ReadImage(filename, &pic);
```

Comment:
You are using the same picture over and over again thus re-writing the memory inside. You need to create a new pic every time thus having a vector of WebPPicture in main.cc. I know it is weird but the main thread should keep all the images in memory until GenerateAnimation is called.
This is because we want to prevent too many image copies over the place (it is slow ....)

Reply:
You now have a memory leak: your pics are created but never freed. Store them in a vector of unique_ptr so that they are magically deleted at the end.